Sasquatch/Sasquatch/ViewControllers/MSSignInViewController.swift

```diff
@@ -0,0 +1,158 @@
+import UIKit
+import WebKit
+
+class MSSignInViewController: UIViewController, WKNavigationDelegate {
+
+  var onAuthDataRecieved: ((_ token: String, _ userId: String, _ expiresAt: Date) -> Void)?
+
+  enum AuthAction {
+    case login, signout
+  }
+
+  var webView: WKWebView!
+
+  let baseUrl = "https://login.live.com/oauth20_"
+  let redirectEndpoint = "desktop.srf"
+  let authorizeEndpoint = "authorize.srf"
+  let tokenEndpoint = "token.srf"
+  let signOutEndpoint = "logout.srf"
+  let clientIdParam = "&client_id=000000004C1D3F6C"
+  var redirectParam = "redirect_uri=https://login.live.com/oauth20_desktop.srf".addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)!
+  var refreshParam = "&grant_type=refresh_token&refresh_token="
+  let refreshTokenParam = "refresh_token"
+  lazy var scopeParam = "&scope=service::events.data.microsoft.com::MBI_SSL"
+  var refreshToken = ""
```

Comment:
Just as an FYI:
the property definitions used to be way prettier but because we need to use Swift 3, we can't use some of the nice Swift 4 awesomeness.